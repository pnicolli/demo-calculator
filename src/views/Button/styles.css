.button {
    background-color: #DDDDDD;
    border-radius: 3px;
    margin-top: 5%;
    flex: 1 1 auto;
    font-size: 32px;
}

.button.double {
    flex: 2 1 auto;
}
