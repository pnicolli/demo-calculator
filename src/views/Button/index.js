import React from 'react';
import { func, string } from 'prop-types';
import './styles.css';

const Button = ({
  onClick,
  text,
}) => (
  <div
    className="button"
    onClick={onClick}
  >
    <span>{text}</span>
  </div>
);

Button.propTypes = {
  onClick: func,
  text: string,
};

export default Button;
