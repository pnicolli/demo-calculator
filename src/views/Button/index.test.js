import React from 'react';
import { shallow } from 'enzyme';
import Button from './index';

const baseProps = {
  onClick: () => {},
  text: '2',
};

it('matches previous snapshot', () => {
  const tree = shallow(<Button {...baseProps} />);
  expect(tree).toMatchSnapshot();
});

it('calls the onClick callback', () => {
  const mock = jest.fn();
  const props = Object.assign({}, baseProps, { onClick: mock });
  const comp = shallow(<Button {...props} />);
  comp.simulate('click');
  expect(mock).toHaveBeenCalled();
});
