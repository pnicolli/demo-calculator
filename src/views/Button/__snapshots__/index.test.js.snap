// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches previous snapshot 1`] = `
<div
  className="button"
  onClick={[Function]}
>
  <span>
    2
  </span>
</div>
`;
